lib/pushmi_pullyu/aip_downloader.rb

```diff
@@ -0,0 +1,147 @@
+require 'pushmi_pullyu/aip_downloader_files_directories'
+require 'pushmi_pullyu/aip_logger'
+require 'pushmi_pullyu/fedora_object_fetcher'
+require 'pushmi_pullyu/solr_fetcher'
+
+# Download all of the metadata/datastreams and associated data
+# related to an object
+
+class PushmiPullyu::AipDownloader
+
+  include PushmiPullyu::AipDownloaderFilesDirectories
+
+  attr_reader :noid, :config, :logger, :aip_logger, :fetcher,
+              :aipcreation_log
+
+  attr_accessor :got_characterization, :got_thumbnail,
+                :got_fedora3foxml, :got_fedora3foxml_metadata
+
+  def initialize(noid, config = nil, application_logger = nil)
+    @noid = noid
+
+    @config = config || PushmiPullyu.options
+    @logger = application_logger || PushmiPullyu.logger
+    @fetcher = PushmiPullyu::FedoraObjectFetcher.new(self.noid, self.config)
+
+    initialize_directory_names
+    initialize_filenames
+
+    # We need to log to application log and to the AIP log
+    @aip_logger = PushmiPullyu::AipLogger.new(@noid, aipcreation_log, @logger)
```

Comment:
That makes sense as far as it goes, but this still needs to be folded into the main logging file. Nothing about that is intended to mean you have to "make two logging statements", there's no reason you can't push the "write two streams" logic into the one logger. Add methods and logic to write to multiple files and anything else you need, but do it in the one place that's related to logging in the codebase, rather than adding a second one.

Think about it from the perspective of a new reader to the codebase: you see a log call, hit cmd-t and start typing "logger" to find the relevant file, and then you discover that there's *multiple* loggers and you get to dig through 2, 3, 7, however many definitions of different loggers, done in different styles, to figure out which one is this one. What do we gain from pushing that confusion on future readers? What do we gain from establishing a pattern of creating different logging classes for every need? Let's keep this codebase as clean and simple as we can make it.